src/TestSuite/IntegrationTestTrait.php

```diff
@@ -876,6 +876,25 @@ public function assertRedirect($url = null, $message = '')
         }
     }
 
+    /**
+     * Asserts that the Location header is correct. Comparison is made against exactly the URL provided.
+     *
+     * @param string|array|null $url The URL you expected the client to go to. This
+     *   can either be a string URL or an array compatible with Router::url(). Use null to
+     *   simply check for the existence of this header.
+     * @param string $message The failure message that will be appended to the generated message.
+     * @return void
+     */
+    public function assertRedirectEquals($url = null, $message = '')
```

Comment:
Should this be `assertRedirectSame()`? PHPUnit uses Same to indicate additional precision/type checks.

Reply:
I took my cues from functions in here like `assertResponseEquals` and helper classes like `BodyEquals`, `HeaderEquals`, etc., but can certainly change it if `Same` is preferred.